bcgov/moh-prime
# PRIME-1859/PRIME-1849 Updates to Site Registration Claim

**Site Registration Updates**
* After claiming an organization the previous signing authority should be forced to fill in their demographic information. 
 So the `PartyEnrolment` needed to be removed during the approval of the claim
* After claiming an organization all site actions were blocked by the provider on each site for that organization not being updated to the new signing authority.  So the `Site.ProvisionerId` needed to be updated during the approval of the claim

**Admin Updates**
* Was possible on claim approval to have a failure (ie. sending email) before the claim was deleted leaving a claim that will never get deleted and remains visible in the web client as each approval attempt indicates the signing authority could not be approved since their `ID` equals the `ID` of the new signing authority.  Refactored the controller and service method to not check the count of updates from the database save and perform ID comparison indicating a notification should be sent or not
* Web client was invoking `getSigningAuthorityById` on every page load even if a new signing authority ID wasn't provided providing `undefined` as a URI param and a failed response
* Web client was invoking `getSigningAuthorityByUserId` when it should have been `getSigningAuthorityById` after receiving the `OrganizationClaim` which contains the ID not a user GUID

**Testing**
1. Create a signing authority and a few sites then logout, remembering the PEC for one of the sites for claiming
2. Create a new signing authority and claim the organization using the PEC then logout
3. Access admin and approve the claim and logout
4. Login as the swapped signing authority from step 2, and you should access site management and be able to visit the overview page, and perform the allowed actions on the site then logout
5. Login as the initial signing authority and you should be forced to enter information into the demographic page
